pkg/network/setup/netconf_test.go

```diff
@@ -66,23 +70,79 @@ var _ = Describe("netconf", func() {
 	})
 
 	It("fails the teardown run", func() {
-		netConf := netsetup.NewNetConfWithCustomFactory(nil, failingCacheCreator{})
+		netConf := netsetup.NewNetConfWithCustomFactoryAndConfigState(nil, failingCacheCreator{}, map[string]*netsetup.ConfigState{})
 		Expect(netConf.Teardown(vmi)).NotTo(Succeed())
 	})
+
+	Context("HotUnplugInterfaces", func() {
+		var shouldFail *bool
+		var wasExecuted *bool
+
+		const netName = "multusNet"
+		const nadName = "blue"
+
+		BeforeEach(func() {
+			dutils.MockDefaultOwnershipManager()
+
+			vmi.Spec.Networks = []v1.Network{{
+				Name: netName,
+				NetworkSource: v1.NetworkSource{
+					Multus: &v1.MultusNetwork{NetworkName: nadName}},
+			}}
+			iface := v1.Interface{
+				Name:                   netName,
+				InterfaceBindingMethod: v1.InterfaceBindingMethod{Bridge: &v1.InterfaceBridge{}},
+			}
+			vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{iface}
+
+			cacheCreator := tempCacheCreator{}
+			shouldFail = pointer.Bool(false)
+			wasExecuted = pointer.Bool(false)
+			nsStub := netnsStub{shouldFail, wasExecuted}
+			factory := func(_ int) netsetup.NSExecutor { return nsStub }
+			configMap := map[string]*netsetup.ConfigState{}
+			configCache := netsetup.NewConfigStateCache(string(vmi.UID), &cacheCreator)
+			configState := netsetup.NewConfigStateWithPodIfaceMap(&configCache, nsStub, launcherPid, map[string]string{netName: "pod1"})
+			configMap[string(vmi.UID)] = &configState
+			netConf = netsetup.NewNetConfWithCustomFactoryAndConfigState(factory, &cacheCreator, configMap)
+
+			podIfaceCacheData := &cache.PodIfaceCacheData{State: cache.PodIfaceNetworkPreparationFinished}
+			Expect(cache.WritePodInterfaceCache(&cacheCreator, string(vmi.UID), netName, podIfaceCacheData)).To(Succeed())
+			vmi.Spec.Domain.Devices.Interfaces[0].State = v1.InterfaceStateAbsent
+		})
+
+		It("runs successfully", func() {
+			Expect(netConf.HotUnplugInterfaces(vmi)).To(Succeed())
+			Expect(*wasExecuted).To(BeTrue())
+		})
+		It("fails the unplug", func() {
+			*shouldFail = true
+
+			Expect(netConf.HotUnplugInterfaces(vmi)).NotTo(Succeed())
+			Expect(*wasExecuted).To(BeTrue())
+		})
+	})
 })
 
 type netnsStub struct {
-	shouldFail bool
+	shouldFail *bool
+	executed   *bool
 }
 
 func (n netnsStub) Do(func() error) error {
-	if n.shouldFail {
+	if n.executed != nil {
+		if *n.executed == true {
+			panic("executed should be false before execution")
+		}
+		*n.executed = true
+	}
+	if n.shouldFail != nil && *n.shouldFail == true {
 		return fmt.Errorf("do-netns failure")
 	}
 	return nil
 }
 func nsNoopFactory(_ int) netsetup.NSExecutor    { return netnsStub{} }
-func nsFailureFactory(_ int) netsetup.NSExecutor { return netnsStub{shouldFail: true} }
+func nsFailureFactory(_ int) netsetup.NSExecutor { return netnsStub{shouldFail: pointer.Bool(true)} }
```

Comment:
I'm not sure if it is necessary to implement these boolean values as pointers, do you think you could explain why you decided to use pointers for the boolean values here?